### PR TITLE
host spec location attributes support

### PR DESCRIPTION
**Pull request by sunilkumarn417**
Signed-off-by: Sunil Kumar Nagaraju <sunnagar@redhat.com>

- provision to support location attributes of the node.
- updated changes required in host spec jinja templates.
- OSD deployment with label option.

**Test results:**
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-WEEA8Q/

Host spec jinja template creation:
```
data:
------


hosts = [
    {
        "hostname": "host-1",
        "address": "10.0.0.1",
    },
    {
        "hostname": "host-2",
        "address": "10.0.0.2",
        "labels": ["label-1", "label-2"],
        "location": {
            "datacenter": "DC1",
            "root": "default"
        }
    },
    {
        "hostname": "host-3",
        "address": "10.0.0.3",
        "labels": ["label-1", "label-2"],
    },
    {
        "hostname": "host-4",
        "address": "10.0.0.4",
        "labels": ["label-3", "label-4"],
        "location": {
            "datacenter": "DC2"
        }
    }
]

print(Template(open("ceph/ceph_admin/jinja_templates/host.jinja").read()).render(hosts=hosts) +
      Template(open("ceph/ceph_admin/jinja_templates/host.jinja").read()).render(hosts=hosts))


❯ python3 /home/sunil/.config/JetBrains/PyCharmCE2022.2/scratches/scratch_2.py
---
service_type: host
hostname: host-1
addr: 10.0.0.1
---
service_type: host
hostname: host-2
addr: 10.0.0.2
labels: ['label-1', 'label-2']
location:
  datacenter: DC1
  root: default
---
service_type: host
hostname: host-3
addr: 10.0.0.3
labels: ['label-1', 'label-2']
---
service_type: host
hostname: host-4
addr: 10.0.0.4
labels: ['label-3', 'label-4']
location:
  datacenter: DC2
---
service_type: host
hostname: host-1
addr: 10.0.0.1
---
service_type: host
hostname: host-2
addr: 10.0.0.2
labels: ['label-1', 'label-2']
location:
  datacenter: DC1
  root: default
---
service_type: host
hostname: host-3
addr: 10.0.0.3
labels: ['label-1', 'label-2']
---
service_type: host
hostname: host-4
addr: 10.0.0.4
labels: ['label-3', 'label-4']
location:
  datacenter: DC2
```

### CODE REVIEW

**Review by psathyan — COMMENTED** [review]
Thanks for putting this together... some not important changes

File: suites/pacific/rados/test-location-stretch-mode.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  - test:
+      name: Install ceph pre-requisites
+      desc: installation of ceph pre-requisites
+      module: install_prereq.py
+      abort-on-fail: true
+  - test:
+      name: Cephadm Bootstrap with apply-spec
+      desc: Apply spec in Bootstrap with host location attributes
+      module: test_bootstrap.py
+      polarion-id:
```
Comment (psathyan) [inline suites/pacific/rados/test-location-stretch-mode.yaml:13]: Sorry... I believe this was missed out in my earlier review... could we add a Polarion ID, please

**Review by psathyan — COMMENTED** [review]
The code looks good and ready to merge... could we add a Polarion ID for deployment?

File: suites/pacific/rados/test-location-stretch-mode.yaml (hunk 1 above)
Comment (sunilkumarn417) [reply suites/pacific/rados/test-location-stretch-mode.yaml]: Updated.

**Review by psathyan — APPROVED** [review]

**Pull request merged**